### ISSUE TITLE
Enum bug in eclipse 4.8.0 (default should be 4.7.3)

### ISSUE THREAD

**Issue by nedtwigg** [diffplug/spotless#263]
I tested out `3.14.0-SNAPSHOT`, which now has @fvgh's great new eclipse framework in place.  It worked great, except for a known bug.  You already pointed it out Frank, but after using it on some large projects, I think we should not make `4.8.0` the default eclipse version.

There is a bug in the `4.8.0` formatter which causes all enums to be indented strangely.  Instead of this:

```java
public enum {
  ONE, TWO;

  public int asNumber() {
    return ordinal();
  }
}
```

We get this:

```java
public enum {
ONE, TWO;

  public int asNumber() {
    return ordinal();
  }
}
```

The bug is tracked at eclipse as [536322](https://bugs.eclipse.org/bugs/show_bug.cgi?id=536322), and it's already fixed for the next release.  Since it's a pretty big behavior change and it's only present in `4.8.0`, I think we should skip it as the default.

Only trouble is, it seems we're missing the lockfile for `4.7.3`, and also for `4.6.2`.  I think we should:

- [ ] add lockfile for `4.6.2` and `4.7.3`
- [ ] change the default to `4.7.3`, with a note that we're skipping `4.8.0` as default because of this issue

Once this is done, I'm ready to cut a new release!  Would you like to merge #241 and #238 first, @fvgh?

**Comment by fvgh** [comment]
No, please continue. Both pending PRs still contain copy&paste of plugin configuration from Eclipse. I [fixed that](https://github.com/diffplug/spotless/commit/28a607344ee4d103448329708bdc5dcb84f86297#diff-3399f69c5790b05bbdc14a5969980e03R146) already for Eclipse-Groovy (GrEclipse). Furthermore I am afraid we still have to make a minor enhancement on lib/lib-extra. I will show you later on. Don't worry, everything I have merged so far is working... (except of this problem :smile: ).

Do you want me to provide the fix for this issue?

Out of curiosity, I used [search.maven.org](https://search.maven.org), to see which old versions of `spotless-ext-eclipse-jdt` are available. `4.6.2` and `4.7.3` didn't and don't show up. Can you recommend another tool to search Maven Central?

**Comment by nedtwigg** [comment]
You are correct that `4.6.2` and `4.7.3` don't exist for `spotless-ext-eclipse-jdt`.  But `goomph` knows about them, so we should be able to make a lockfile for them.  If you don't have time to provide the lockfiles, can you point me to instructions for how to make them?

**Comment by fvgh** [comment]
In this case, maybe I should explain the missing bit in lib/lib-extra. Then you better work on that one. Not much to do, but some one has to determine how to change the interface and I would prefer to leave that to you. One mom., just opening a new PR demonstrating the problem...

**Comment by fvgh** [comment]
Please have a look at #264 and decide whether we need a fix for this release.
You explained how goomph works, but keep in mind that (at least for `4.6.2`) not all required versions might be available on M2. If I shall provide a fix for `4.7.3`, I would basically just change the `org.eclipse.jdt:org.eclipse.jdt.core` version. The Eclipse OSGI versions you can't change anyhow, since `com.diffplug.spotless:spotless-eclipse-base` interface implementation depends on them. But I see not a point to ape Eclipse versions completely. These new versions of OSGI interfaces are backward compatible and have no impact on the formatter result at all. Same is true for core runtime, resources and commands.
